src/common/multi_file_reader.cpp

```diff
@@ -427,16 +427,24 @@ void MultiFileReaderOptions::AutoDetectHiveTypesInternal(const string &file, Cli
 	}
 
 	const LogicalType candidates[] = {LogicalType::DATE, LogicalType::TIMESTAMP, LogicalType::BIGINT};
+	const bool strictness[] = {true, false};
```

Comment:
I think we should always do a strict cast - non-strict cast can be lossy. In case of a lossy cast I would argue we should prefer to use a VARCHAR instead.

Reply:
That's what I did at first, but a bunch of tests started failing.

Reply:
Should those tests have been succeeding in the first place? I would say we should guarantee that this is non-lossy by falling back to a string if we can't guarantee that.

Reply:
OK I'll take another pass, but my morning is a bit full.

Reply:
Turns out I was confused and everything passes with strict.